test/__init__.py

```
@@ -19,6 +28,87 @@
+    cache_root.mkdir(parents=True, exist_ok=True)
+    with open(cache_root / f"{target_dir.name}.lock", "w") as lock_file:
+        fcntl.flock(lock_file, fcntl.LOCK_EX)
+        try:
+            if marker.exists():  # another process installed while we were waiting for the lock
+                return
+            # A directory left behind by a crashed install is missing the marker, so it
+            # would reach this point again -- but pip's --target only reliably behaves
+            # against an empty directory, so a stale partial install must be cleared first.
+            shutil.rmtree(target_dir, ignore_errors=True)
+            target_dir.mkdir(parents=True, exist_ok=True)
+            subprocess.run(
```

Ruff (0.16.1)

[error] `subprocess` call: check for execution of untrusted input

(S603)